sound/soc/sof/ipc4-topology.c

```diff
@@ -1975,6 +2015,9 @@ static int sof_ipc4_route_setup(struct snd_sof_dev *sdev, struct snd_sof_route *
 		return sroute->src_queue_id;
 	}
 
+	if (sroute->src_queue_id > 0 && WIDGET_IS_COPIER(src_widget->id))
+		sof_ipc4_set_copier_sink_format(sdev, src_widget, sroute->src_queue_id);
```

Comment:
In my opinion, we need to first bind, then set sink format since the bind ipc msg will allocate queue buffer in FW and after this the format setting can take effect.

Reply:
> In my opinion, we need to first bind, then set sink format since the bind ipc msg will allocate queue buffer in FW and after this the format setting can take effect.

Agree, and updated.